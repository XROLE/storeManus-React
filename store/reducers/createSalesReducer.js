import {

  CREATE_SALES_IN_PROGRESS,
  CREATE_SALES_SUCCESS,
  CREATE_SALES_ERROR,
} from '../actions/actionTypes';

const initialState = {
  pending: false,
  error: '',
  success: false,
  message: '',
};

export default function (state = initialState, action) {
  switch (action.type) {
    case CREATE_SALES_IN_PROGRESS: {
      return {
        ...state,
        pending: true,
      };
    }
    case CREATE_SALES_SUCCESS: {
      return {
        ...state,
        pending: false,
        success: true,
        sales: action.payload.sale,
      };
    }
    case CREATE_SALES_ERROR: {
      return {
        ...state,
        error: action.error,
        pending: false,
      };
    }
    default:
      return state;
  }
}
