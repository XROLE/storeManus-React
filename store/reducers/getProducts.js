import {
  GET_PRODUCT_IN_PROGRESS,
  GET_PRODUCT_SUCCESS,
  GET_PRODUCT_ERROR,
} from '../actions/actionTypes';

const initialState = {
  message: '',
  pending: false,
  error: '',
  products: '',
};

export default function (state = initialState, action) {
  switch (action.type) {
    case GET_PRODUCT_IN_PROGRESS: {
      return {
        ...state,
        error: null,
        pending: true,
      };
    }
    case GET_PRODUCT_SUCCESS: {
      return {
        ...state,
        pending: false,
        products: action.payload.products,
      };
    }
    case GET_PRODUCT_ERROR: {
      return {
        ...state,
        error: action.error,
        pending: false,
      };
    }
    default:
      return state;
  }
}
