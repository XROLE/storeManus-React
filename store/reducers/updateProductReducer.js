import {
  UPDATE_PRODUCT_IN_PROGRESS,
  UPDATE_PRODUCT_SUCCESS,
  UPDATE_PRODUCT_ERROR,
} from '../actions/actionTypes';

const initialState = {
  message: '',
  pending: false,
  error: '',
  updatedAvailableProduct: '',
  success: false,
};

export default function (state = initialState, action) {
  switch (action.type) {
    case UPDATE_PRODUCT_IN_PROGRESS: {
      return {
        ...state,
        error: null,
        pending: true,
      };
    }
    case UPDATE_PRODUCT_SUCCESS: {
      return {
        ...state,
        pending: false,
        success: true,
        updatedAvailableProduct: action.payload.editedProduct,
      };
    }
    case UPDATE_PRODUCT_ERROR: {
      return {
        ...state,
        error: action.error,
        pending: false,
      };
    }
    default:
      return state;
  }
}
