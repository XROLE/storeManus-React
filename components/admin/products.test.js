import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import axios from 'axios';
import { BrowserRouter } from 'react-router-dom';
import Products from './Products';
import getProducts from '../../store/reducers/getProducts';
import updateProductReducer from '../../store/reducers/updateProductReducer';
import createSalesReducer from '../../store/reducers/createSalesReducer';

jest.mock('axios');

describe('Product Products', () => {
  it('Should render just product component', () => {
    const component = shallow(
      <Products />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});

describe('Products Component', () => {
  const get = jest.fn();
  const initialState = {
    getProduct: {
      message: '',
      pending: false,
      error: '',
      products: '',
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore(() => initialState);
  const dispatchMock = jest.spyOn(store, 'dispatch');
  const pending = false;
  const products = [];

  const props = {
    Productss: {
      products,
      pending,
      get,
    },
  };

  axios.get.mockImplementation(() => Promise.resolve({
    data: { products: { name: 'Salt' } },
  }));

  mount(
    <Provider store={store}>
      <BrowserRouter>
        <Products {...props} />
      </BrowserRouter>
    </Provider>,
  );

  expect(dispatchMock).toHaveBeenCalled();
});

describe('Products Component', () => {
  const get = jest.fn();
  const initialState = {
    getProduct: {
      message: '',
      pending: false,
      error: '',
      products: '',
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore(() => initialState);
  const dispatchMock = jest.spyOn(store, 'dispatch');
  const pending = false;
  const products = [];

  const props = {
    Productss: {
      products,
      pending,
      get,
    },
  };

  const error = {
    response: {
      data: {
        Message: 'An error occured',
      },
    },
  };
  axios.get.mockImplementation(() => Promise.reject(error));

  mount(
    <Provider store={store}>
      <BrowserRouter>
        <Products {...props} />
      </BrowserRouter>
    </Provider>,
  );

  expect(dispatchMock).toHaveBeenCalled();
});


it('should setup default state values', () => {
  const state = getProducts(undefined, {
    type: '@@INIT',
  });
  expect(state).toEqual({
    message: '',
    pending: false,
    error: '',
    products: '',
  });
});

it('should update state when action type is  GET_PRODUCT_IN_PROGRESS', () => {
  const state = getProducts(undefined, {
    type: 'GET_PRODUCT_IN_PROGRESS',
    pending: true,
  });
  expect(state).toEqual({
    ...state,
    pending: true,
  });
});

it('should update state when action type is  GET_PRODUCT_SUCCESS', () => {
  const state = getProducts(undefined, {
    type: 'GET_PRODUCT_SUCCESS',
    payload: {
      products: '',
    },
  });
  expect(state).toEqual({
    ...state,
    products: '',
  });
});

it('should update state when action type is  GET_PRODUCT_ERROR', () => {
  const state = getProducts(undefined, {
    type: 'GET_PRODUCT_ERROR',
    error: 'failed',
  });
  expect(state).toEqual({
    ...state,
    error: 'failed',
    pending: false,
  });
});


it('should setup default state values', () => {
  const state = getProducts(undefined, {
    type: '@@INIT',
  });
  expect(state).toEqual({
    message: '',
    pending: false,
    error: '',
    products: '',
  });
});

it('should update state when action type is  GET_PRODUCT_IN_PROGRESS', () => {
  const state = getProducts(undefined, {
    type: 'GET_PRODUCT_IN_PROGRESS',
    pending: true,
  });
  expect(state).toEqual({
    ...state,
    pending: true,
  });
});

it('should update state when action type is  GET_PRODUCT_SUCCESS', () => {
  const state = getProducts(undefined, {
    type: 'GET_PRODUCT_SUCCESS',
    payload: {
      products: '',
    },
  });
  expect(state).toEqual({
    ...state,
    products: '',
  });
});

it('should update state when action type is  GET_PRODUCT_ERROR', () => {
  const state = getProducts(undefined, {
    type: 'GET_PRODUCT_ERROR',
    error: 'failed',
  });
  expect(state).toEqual({
    ...state,
    error: 'failed',
    pending: false,
  });
});

describe('Update Products reducesr', () => {
  it('should setup default state values', () => {
    const state = updateProductReducer(undefined, {
      type: '@@INIT',
    });
    expect(state).toEqual({
      message: '',
      pending: false,
      error: '',
      success: false,
      updatedAvailableProduct: '',
    });
  });

  it('should update state when action type is  UPDATE_PRODUCT_IN_PROGRESS', () => {
    const state = updateProductReducer(undefined, {
      type: 'UPDATE_PRODUCT_IN_PROGRESS',
      error: null,
      pending: true,
    });
    expect(state).toEqual({
      ...state,
      error: null,
      pending: true,
    });
  });

  it('should update state when action type is  UPDATE_PRODUCT_SUCCESS', () => {
    const state = updateProductReducer(undefined, {
      type: 'UPDATE_PRODUCT_SUCCESS',
      payload: {
        success: true,
      },
    });
    expect(state).toEqual({
      ...state,
      success: true,
    });
  });

  it('should update state when action type is  UPDATE_PRODUCT_ERROR', () => {
    const state = updateProductReducer(undefined, {
      type: 'UPDATE_PRODUCT_ERROR',
      error: '',
      pending: false,
    });
    expect(state).toEqual({
      ...state,
      error: '',
      pending: false,
    });
  });
});

describe('Update Create Sales reducesr', () => {
  it('should setup default state values', () => {
    const state = createSalesReducer(undefined, {
      type: '@@INIT',
    });
    expect(state).toEqual({
      pending: false,
      error: '',
      success: false,
      message: '',
    });
  });

  it('should update state when action type is  CREATE_SALES_IN_PROGRESS', () => {
    const state = createSalesReducer(undefined, {
      type: 'CREATE_SALES_IN_PROGRESS',
      pending: true,
    });
    expect(state).toEqual({
      ...state,
      pending: true,
    });
  });

  it('should update state when action type is  CREATE_SALES_SUCCESS', () => {
    const state = createSalesReducer(undefined, {
      type: 'CREATE_SALES_SUCCESS',
      payload: {
        pending: false,
        success: true,
      },
    });
    expect(state).toEqual({
      ...state,
      success: true,
    });
  });

  it('should update state when action type is  CREATE_SALES_ERROR', () => {
    const state = createSalesReducer(undefined, {
      type: 'CREATE_SALES_ERROR',
      error: null,
      pending: false,
    });
    expect(state).toEqual({
      ...state,
      error: null,
      pending: false,
    });
  });
});
