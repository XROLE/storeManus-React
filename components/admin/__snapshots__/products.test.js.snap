// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Products Should render just product component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
